### PR TITLE
Merge tsdb into prometheus

### PR DESCRIPTION
**Pull request by codesome**
**DO NOT SQUASH IF MERGING**

### Few reason for merge

* Keeping Prometheus and tsdb in sync, versioning etc, is a pain (quoted from dev summit notes)
* Running prombench also becomes tedious when the changes are in tsdb, as we have to open a PR in this repo and keep in sync with the tsdb PR.
* There was consensus to merge tsdb with Prometheus in the 2019 dev summit 1 at Barcelona (https://docs.google.com/document/d/1NQIX78nwBhfLZD3pAb0PK-uBKYqnkzjjVhOQ-kIaEGU/edit#)

### In this PR

I followed this blog as is https://thoughts.t37.net/merging-2-different-git-repositories-without-losing-your-history-de7a06bba804.

All commit history seems to be preserved.

- [x] Change all the imports and don't import tsdb
- [x] Fix the vendoring of tsdb.
- [x] Merge the tests and travis/circle-ci configs.
- [x] tsdb `Makefile`

### After merging

1. For transferring issues: https://help.github.com/en/articles/transferring-an-issue-to-another-repository
2. For transferring PRs: Author of the PR would need to do it manually.

### CODE REVIEW

File: tsdb/.github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<!--
```
Comment (bwplotka) [inline tsdb/.github/PULL_REQUEST_TEMPLATE.md]: I think we need to merge this with Prometheus one.

File: tsdb/.travis.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+dist: trusty
```
Comment (bwplotka) [inline tsdb/.travis.yml]: ditto

**Comment by codesome** [comment]
TODO: Change all the imports and don't import tsdb (After an LGTM)

**Comment by codesome** [comment]
This is ready for another round of review. Can't really help the DCO failure because commits are from the time when DCO was still not enforced.

**Comment by brian-brazil** [comment]
:+1: 

I don't see anything obviously wrong here.

**Comment by beorn7** [comment]
As annoying as it is at this stage, but perhaps we should wait for the outcome of https://groups.google.com/forum/#!topic/prometheus-developers/F1Vp0rLk3TQ before merging this.

**Comment by bwplotka** [comment]
@codesome can we rebase it? Looks like we are happy to potentially merge it before next release tomorrow. I think we need to include any new commit. Happy to help as well if you are buysy

**Comment by bwplotka** [comment]
@juliusv https://matrix.to/#/!WaUKIfoqfiyWQhenET:matrix.org/$1565681997305908PSsSd:matrix.org?via=matrix.org&via=half-shot.uk

**Comment by codesome** [comment]
I have rebased with the current tsdb master. I have to verify if all the cleanup that I had done are still the same.

**Comment by bwplotka** [comment]
```
--- FAIL: TestFileSD (16.67s)
    file_test.go:113: Unexpected target groups [], we expected a failure here.
FAIL
```
Flake?

**Comment by codesome** [comment]
> Flake?

I think so. I restarted the test and it passed.

**Comment by gouthamve** [comment]
Haven't been following this too closely, but agree with the merge overall. Lets do ittttt! :+1:

**Comment by beorn7** [comment]
Let's merge on green.

**Comment by bwplotka** [comment]
Can we merge this to ensure README has the info? https://github.com/prometheus/tsdb/pull/677

**Comment by bwplotka** [comment]
All issues were transferred to Prometheus with label `components/tsdb`

**Pull request merged**